CorsixTH/Lua/world.lua

```diff
@@ -1090,15 +1090,35 @@ function World:onTick()
 end
 
 function World:setEndMonth()
+  local previous_date = self.game_date
   local first_day_of_next_month = Date(self.game_date:year(), self.game_date:monthOfYear() + 1)
   self.game_date = first_day_of_next_month:plusHours(-1)
+  -- Has the date jump caused an emergency to be missed?
+  if self:wasEmergencySkipped(previous_date, self.game_date) then
+    self:nextEmergency()
+  end
 end
 
 function World:setEndYear()
+  local previous_date = self.game_date
   local first_day_of_next_year = Date(self.game_date:year() + 1)
   self.game_date = first_day_of_next_year:plusHours(-1)
+  -- Has the date jump caused an emergency to be missed?
+  if self:wasEmergencySkipped(previous_date, self.game_date) then
+    self:nextEmergency()
+  end
```

Comment:
What happens if you missed more than one emergency?

Doing "nextEmergency" here exposes internal details of the emergency code, a cleaner solution would be to write a `skipSemergenciesTo(new_date)`, other option is to exent the existing function to `self:nextEmergency(new_date)`

Reply:
After 1843 the code will skip to the next emergency in the list. It's more kind to skip just the missed emergency than skip all emergencies for e.g. a whole year.

Not sure what you mean by exposing internals.

Reply:
Actually just to check, this resolves itself from changes in 1843. It'll autoskip.
```
<<Afterload in 1843>>
Emergency       2        skipped
Emergency       3        skipped
Emergency       4        skipped
Emergency       5        skipped
Emergency       6        skipped
Next emergency is       2-12-09T00
Next emergency is       2-12-03T00
Next emergency is       3-02-13T00
<<12 month jump>>
Emergency       10       skipped
Emergency       11       skipped
Emergency       12       skipped
Next emergency is       3-12-31T00
Next emergency is       3-12-04T00
Next emergency is       4-02-06T00
New emergency set!
```